rebasehelper/plugins/spec_hooks/replace_old_version.py

```diff
@@ -114,18 +99,33 @@ def run(cls, spec_file: SpecFile, rebase_spec_file: SpecFile, **kwargs: Any):
         replace_with_macro = bool(kwargs.get('replace_old_version_with_macro'))
 
         subversion_patterns = cls._create_possible_replacements(old_version, new_version, replace_with_macro)
+        examined_lines = collections.defaultdict(set)
+        for tag in rebase_spec_file.tags.items:
```

Comment:
First, `.items` is completely unnecessary and shouldn't be used. Second, you are now including also invalid tags. I would use `rebase_spec_file.tags.filter(section=None)`.